From 049d3ce1efda40a59bd42834d3b8db5cdb0386fa Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 6 Jun 2011 11:39:45 +0000
Subject: [PATCH 1/3] setup-xorg-base: initial commit

This script is supposed to help users to quickly get a working xorg server
running.
---
 .gitignore         |    1 +
 Makefile           |    1 +
 setup-xorg-base.in |   29 +++++++++++++++++++++++++++++
 3 files changed, 31 insertions(+), 0 deletions(-)
 create mode 100644 setup-xorg-base.in

diff --git a/.gitignore b/.gitignore
index c1f3e8e..e0b22b1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -16,4 +16,5 @@ setup-interfaces
 setup-keymap
 setup-mta
 setup-timezone
+setup-xorg-base
 update-conf
diff --git a/Makefile b/Makefile
index 96ec9cc..91f0835 100644
--- a/Makefile
+++ b/Makefile
@@ -24,6 +24,7 @@ SBIN_FILES	:= lbu\
 		setup-acf\
 		setup-bootable\
 		setup-timezone\
+		setup-xorg-base\
 		update-conf
 
 BIN_FILES	:= uniso
diff --git a/setup-xorg-base.in b/setup-xorg-base.in
new file mode 100644
index 0000000..bcdbe77
--- /dev/null
+++ b/setup-xorg-base.in
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+# simple script to setup x basic org server
+# you will still need a window manager and login manager or xinit
+
+base_pkgs="xorg-server xf86-video-vesa xf86-input-evdev xf86-input-mouse
+	xf86-input-keyboard udev"
+
+# TODO: detect graphics card and pick proper xf86-video-* driver based on that
+
+# TODO: detect if we need xf86-input-synaptics
+
+# ps mouse
+modprobe psmouse
+grep -q -w psmouse /etc/modules || echo "psmouse" >> /etc/modules
+
+# install packages
+apk add $base_pkgs $@
+
+# setup and start udev
+rc-update -q del mdev sysinit
+rc-update -q add udev sysinit
+rc-update -q add udev-postmount default
+
+if ! rc-service -q udev status; then
+	rc-service udev start
+	rc-service udev-postmount start
+fi
+
-- 
1.7.5.4

